app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -160,6 +167,7 @@ class AppsPreferences(private val prefs: SharedPreferences) {
         return when {
             pref.toLowerCase() == "ssh" || (app.supportsCli && !app.supportsGui) -> SshTypePreference
             pref.toLowerCase() == "vnc" || (!app.supportsCli && app.supportsGui) -> VncTypePreference
+            pref.toLowerCase() == "xsdl" || (!app.supportsCli && app.supportsGui) -> XsdlTypePreference
```

Comment:
The second half of the conditional here will never be hit, as it would already be true for the expression in the line above.

Reply:
I've removed the second half of the conditional, I can understand why it would not be hit there

Reply:
Wouldn't this never hit the third condition?

Reply:
If you're referring to `app.supportsGui` as the third condition then yes it would never hit that part.  So I removed that part too

[The line is now reduced to](https://github.com/CypherpunkArmory/UserLAnd/pull/578/files#diff-97f43415e5835dd28a9d0d0c0387503eR170):
`pref.toLowerCase() == "xsdl" -> XsdlTypePreference`

Reply:
Oh, @AndrewScibek  is actually mostly right. The third condition is only ever hit if `prefs.toLowerCase() == "vnc"` is false *and* (!app.supportsCli && app.supportsGui)` is false. 

I think you could probably leave out your conditional completely. The expression is meant to allow us to default to either SSH if we don't support a graphical environment for an app, or VNC if we don't support a terminal for it. We'd need some kind of top-level user setting to determine if they wanted stuff like this to default to VNC or X.

Reply:
Additionally, given that XSDL doesn't have support for Android 9 right now, I think the safe bet is to just default to VNC. Saves the overhead of doing extra version guarding there as well.